File: apps/flask/modules/node.py
class Node:
	def __init__(self, value):
		self.value = value 
		self.children = []
		self.path = set()
		
	def add_visited(self, node):
		self.visited.add(node)
	
	def add_child(self, child_node):
		self.children.append(child_node)
		
	def set_path(self, node_path):
		if not self.path or len(node_path) < len(self.path):
			self.path = node_path
		
	def remove_child(self, child_node):
		self.children.discard(child_node)
		

File: libs/react-js/components/node.js
export default (value) => {
    return {
        value: value,
        children: [],
        path: new Set(),

        addVisited(node) {
            this.path.add(node);
        },

        addChild(childNode) {
            this.children.push(childNode);
        },

        setPath(nodePath) {
            if (this.path.size == 0 || nodePath.length < this.path.size) this.path = nodePath;
        },

        removeChild(childNode) {
            this.children = this.children.filter((i)=> i != childNode);
        }
    }
}